### PR TITLE
Enhancement/ciac 10555/ews o365 gcc high support

### PR DESCRIPTION
**Pull request by barryyosi-panw**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [CIAC-10555](https://jira-dc.paloaltonetworks.com/browse/CIAC-10555)

## Description
Support for difference azure cloud environments.

[endpoints ref](https://learn.microsoft.com/en-us/microsoft-365/enterprise/)

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py [hunk 1]
```diff
@@ -127,6 +127,31 @@
 """ Classes """
 
 
+class CustomDomainOAuth2Credentials(OAuth2AuthorizationCodeCredentials):
```
Comment (barryyosi-panw) [inline Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py:130]: [Solution refernce](https://github.com/ecederstrand/exchangelib/issues/1317)

File: Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py [hunk 2]
```diff
@@ -415,6 +420,8 @@ def __init__(self,
         synapse_analytics_resource_id='https://dev.azuresynapse.usgovcloudapi.net',
         portal='https://portal.azure.us',
         keyvault='https://vault.usgovcloudapi.net',
+        exchange_online='https://outlook-dod.office365.us'
```
Comment (barryyosi-panw) [inline Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py:423]: [endpoint ref](https://learn.microsoft.com/en-us/microsoft-365/enterprise/microsoft-365-u-s-government-dod-endpoints?view=o365-worldwide)

File: Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py [hunk 3]
```diff
@@ -481,6 +489,7 @@ def __init__(self,
         synapse_analytics_resource_id='https://dev.azuresynapse.azure.cn',
         portal='https://portal.azure.cn',
         keyvault='https://vault.azure.cn',
+        exchange_online='https://partner.outlook.cn'
```
Comment (barryyosi-panw) [inline Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py:492]: [endpoint ref](https://learn.microsoft.com/en-us/microsoft-365/enterprise/urls-and-ip-address-ranges-21vianet?view=o365-worldwide&viewFallbackFrom=o365-worldwid#:~:text=No-,partner.outlook.cn,-40.73.132.0/24%2C%2040.73.164.128)

File: Packs/MicrosoftExchangeOnline/Integrations/EWSO365/README.md [hunk 4]
```diff
@@ -37,6 +37,7 @@ The EWS integration can be used for the following use cases.
 2.  Search for EWS O365.
 3.  Click **Add instance** to create and configure a new integration instance.
     *   **Name**: a textual name for the integration instance.
+    *   **Azure Cloud**: Azure Cloud environment.
```
Comment (michal-dagan) [inline Packs/MicrosoftExchangeOnline/Integrations/EWSO365/README.md:40]:
```suggestion
    *   **Azure Cloud**: Azure Cloud environment. See table below.
```

**Review by michal-dagan — APPROVED** [review]
Great work!!

**Comment by barryyosi-panw** [comment]
Hi @DeanArbel, force merge is required due to invalid grant (bad creds) in Microsoft TPBs during authorization which fails the build.

**Pull request merged**